controllers/flinkcluster_reconciler.go

```diff
@@ -413,37 +413,69 @@ func (reconciler *ClusterReconciler) reconcileJob() (ctrl.Result, error) {
 	var desiredJob = reconciler.desired.Job
 	var observed = reconciler.observed
 	var observedJob = observed.job
+	var recordedJobStatus = observed.cluster.Status.Components.Job
+	var activeFlinkJob bool
 	var err error
 
 	// Update status changed via job reconciliation.
 	var newSavepointStatus *v1beta1.SavepointStatus
 	var newControlStatus *v1beta1.FlinkClusterControlStatus
 	defer reconciler.updateStatus(&newSavepointStatus, &newControlStatus)
 
-	// Create
-	if desiredJob != nil && observedJob == nil {
-		// Proceed job creation process when the cluster components and Flink API server is ready
-		if !isFlinkAPIReady(observed) {
-			log.Info("Waiting for cluster components and Flink API server to be ready")
+	// Cancel unexpected jobs
+	if len(observed.flinkJobStatus.flinkJobsUnexpected) > 0 {
+		log.Info("Cancelling unexpected running job(s)")
+		err = reconciler.cancelUnexpectedJobs(false /* takeSavepoint */)
+		return requeueResult, err
+	}
+
+	// Check if Flink job is active
+	if isJobActive(recordedJobStatus) {
+		activeFlinkJob = true
+	} else {
+		activeFlinkJob = false
+	}
+
+	// Create Flink job submitter
+	if desiredJob != nil && !activeFlinkJob {
+		if !isClusterUpdated(observed) {
```

Comment:
Do you want to express `if isClusterUpdating(...)`? I feel `!isClusterUpdated(...)` is a little confusing, consider changing it to `if isClusterUpdating(...)`.

Reply:
This function is used in several places with a slightly different meaning. Would it be okay to comment instead?